### DISCUSSION TITLE
HashSet<Point> is slower than HashSet<String>

### DISCUSSION THREAD

**Discussion by ericwood8** [dotnet/csharplang#8458] [Language Ideas]
Summary
Fix the Point datatype to be one of the generics throughout all generics in C#.

Motivation
GOAL 1- Performance.
GOAL 2- Less code by developer.

Planning:
1) Point needs to be considered one of the generics.  So check not only HashSet but other things you can set to generics.
2) PointComparer : IEqualityComparer<Point> needs a good Equals() and GetHashCode()

Suggested reading:
https://stackoverflow.com/questions/46142734/why-is-hashsetpoint-so-much-slower-than-hashsetstring

**Comment by CyrusNajmabadi** [comment]
This is not a language concern. This is a runtime concern.

**Comment by koszeggy** [comment]
That SO question is quite outdated. Starting with .NET Core 2.0 `Point` [implements](https://learn.microsoft.com/en-us/dotnet/api/system.drawing.point?view=netcore-2.0) the `IEquatable<T>` interface.
